skbase/base/_tagmanager.py

```diff
@@ -0,0 +1,215 @@
+            Name of flag value.
+        flag_value_default : any type
+            Default/fallback value if flag is not found.
+        flag_attr_name : str, optional, default = "_flags"
+            name of the flag attribute that is read
+
+        Returns
+        -------
+        flag_value :
+            Value of `flag_name` flag in self. If not found, `flag_value_default`.
+        """
+        collected_flags = cls._get_class_flags(flag_attr_name=flag_attr_name)
+
+        return collected_flags.get(flag_name, flag_value_default)
+
+    def _init_flags(self, flag_attr_name="_flags"):
+        """Create dynamic flag dictionary in self.
+
+        Should be called in __init__ of the host class.
+        Creates attribute [flag_attr_name]_dynamic containing an empty dict.
+
+        Parameters
+        ----------
+        flag_attr_name : str, optional, default = "_flags"
+            name of the flag attribute that is read
+
+        Returns
+        -------
+        self : reference to self
+        """
+        setattr(self, f"{flag_attr_name}_dynamic", {})
+        return self
+
+    def _get_flags(self, flag_attr_name="_flags"):
+        """Get flags from estimator class and dynamic flag overrides.
+
+        Parameters
+        ----------
+        flag_attr_name : str, optional, default = "_flags"
+            name of the flag attribute that is read
+
+        Returns
+        -------
+        collected_flags : dict
+            Dictionary of flag name : flag value pairs. Collected from flag_attr_name
+            class attribute via nested inheritance and then any overrides
+            and new flags from [flag_attr_name]_dynamic object attribute.
+        """
+        collected_flags = self._get_class_flags(flag_attr_name=flag_attr_name)
+
+        if hasattr(self, f"{flag_attr_name}_dynamic"):
+            collected_flags.update(getattr(self, f"{flag_attr_name}_dynamic"))
+
+        return deepcopy(collected_flags)
+
+    def _get_flag(
+        self,
+        flag_name,
+        flag_value_default=None,
+        raise_error=True,
+        flag_attr_name="_flags",
+    ):
+        """Get flag value from estimator class and dynamic flag overrides.
+
+        Parameters
+        ----------
+        flag_name : str
+            Name of flag to be retrieved
+        flag_value_default : any type, optional; default=None
+            Default/fallback value if flag is not found
+        raise_error : bool
+            whether a ValueError is raised when the flag is not found
+        flag_attr_name : str, optional, default = "_flags"
+            name of the flag attribute that is read
+
+        Returns
+        -------
+        flag_value :
+            Value of the `flag_name` flag in self. If not found, returns an error if
+            raise_error is True, otherwise it returns `flag_value_default`.
+
+        Raises
+        ------
+        ValueError if raise_error is True i.e. if flag_name is not in self.get_flags(
```

Comment:
Move the description of when the ValueError is raised to the line below the error type (and indent like other descriptions).

Reply:
ok